framework/deproxy_client.py

```diff
@@ -660,6 +656,34 @@ def handle_read(self):
             )
             raise
 
+    def handle_write(self):
+        """
+        Send data from `self.request_buffers` and cut them.
+        Move data from `self.req_body_buffers` to `self.request_buffers`
+            when `self.request_buffers` is empty for current request.
+        Increase `self.cur_req_num` when two buffers are empty for current request.
+        Does not send data when flow_control_window is 0.
+        """
+        cur_req_num = self.cur_req_num
+
+        if self.request_buffers[cur_req_num]:
+            super().handle_write()
+
+        body = self.req_body_buffers[cur_req_num]["body"]
+        if self.request_buffers[cur_req_num] == b"" and body:
+            stream_id = self.req_body_buffers[cur_req_num]["stream_id"]
+            end_stream = self.req_body_buffers[cur_req_num]["end_stream"]
+            if self.cur_req_num > cur_req_num:
```

Comment:
I don't understand case when we can have  self.cur_req_num > cur_req_num here. Can you please explain it in comment

Reply:
I added comment. This happens when both buffers contain data. For example, you use:
```python3
client.make_request(
    client.create_request(
         method="POST", headers=[], body="The request body",
    )
)
```
The headers will be written to the socket at first call of `handle_write` method (`self.cur_req_num` will be increase in `super().handle_write()`), but body will be sent separately in the next call of `handle_write`.